src/serac/physics/utilities/finite_element_state.hpp

```diff
@@ -241,23 +247,175 @@ class FiniteElementState {
    * in a restart run
    * @note Must be const as FESpaces store a const reference to their FEColls
    */
-  MaybeOwningPointer<const mfem::FiniteElementCollection> coll_;
+  detail::MaybeOwningPointer<const mfem::FiniteElementCollection> coll_;
   /**
    * @brief Possibly-owning handle to the FiniteElementCollection, as it is owned
    * by the FiniteElementState in a normal run and by the MFEMSidreDataCollection
    * in a restart run
    */
-  MaybeOwningPointer<mfem::ParFiniteElementSpace> space_;
+  detail::MaybeOwningPointer<mfem::ParFiniteElementSpace> space_;
   /**
    * @brief Possibly-owning handle to the ParGridFunction, as it is owned
    * by the FiniteElementState in a normal run and by the MFEMSidreDataCollection
    * in a restart run
    */
-  MaybeOwningPointer<mfem::ParGridFunction> gf_;
-  mfem::HypreParVector                      true_vec_;
-  std::string                               name_ = "";
+  detail::MaybeOwningPointer<mfem::ParGridFunction> gf_;
+  mfem::HypreParVector                              true_vec_;
+  std::string                                       name_ = "";
 };
 
+/**
+ * @brief A shim class for describing the interface of something that can be synced
+ */
+class SyncableData {
```

Comment:
Am I correct in thinking this is an interface for "history" data with various views, e.g. friction data on a surface?

Reply:
I've expanded on the description/Doxygen documentation for this class to hopefully better capture its raison d'etre.